fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/FederationResult.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.dedis.popstellar.model.network.method.message.data.federation
+
+import com.github.dedis.popstellar.model.network.method.message.MessageGeneral
+import com.github.dedis.popstellar.model.network.method.message.data.Action
+import com.github.dedis.popstellar.model.network.method.message.data.Data
+import com.github.dedis.popstellar.model.network.method.message.data.Objects
+import com.google.gson.annotations.SerializedName
+
+/** Informs about the result of the authentication procedure */
+class FederationResult
+/**
+ * Constructor for a data Federation Result
+ *
+ * @param status status of the result (either success or failure)
+ * @param reason reason of the failure
+ * @param publicKey public key of the other LAO organizer
+ * @param challenge challenge used to connect the LAOs
+ */
+(
+    val status: String,
+    val reason: String? = null,
+    @SerializedName("public_key") val publicKey: String? = null,
+    val challenge: MessageGeneral,
+) : Data {
+
+  init {
+    when (status) {
+      "failure" -> {
+        require(reason != null) { "Reason must be provided for failure status." }
+        require(publicKey == null) { "Public key must be null for failure status." }
+      }
+      "success" -> {
```

Comment:
Perhaps it would be better to define these ("failure" and "success") as static constants, so in case we want to change them in future it's just 1 place to modify

Reply:
Ok, I'll change that